### ISSUE TITLE
Android: Failure during map updates

### ISSUE THREAD

**Issue by dbf256** [omapsapp/omapsapp#65]
1. I installed beta version, it downloaded some maps, but also it was a red dot in menu that suggested to download maps.
2. I pressed download and app closed
3. Now it closes every time I try to open menu. Stacktrace for failure it:

```
Build fingerprint: 'Xiaomi/raphael_eea/raphael:9/PKQ1.181121.001/V10.3.3.0.PFKEUXM:user/release-keys'
Revision: '0'
ABI: 'arm64'
pid: 23131, tid: 23131, name: app.omaps.beta  >>> app.omaps.beta <<<
signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
    x0  0000000000000000  x1  0000000000005a5b  x2  0000000000000006  x3  0000000000000008
    x4  00000000000024e0  x5  00000000000024e0  x6  00000000000024e0  x7  0000000000000018
    x8  0000000000000083  x9  c752345313dfa34c  x10 0000000000000000  x11 fffffffc7ffffbdf
    x12 0000000000000001  x13 0000000000000009  x14 000000007005c980  x15 000000000000fbdc
    x16 000000799326f2b8  x17 000000799318eee0  x18 0000000000000008  x19 0000000000005a5b
    x20 0000000000005a5b  x21 0000000000000001  x22 0000007997ca75e0  x23 00000078f7d862c1
    x24 0000000000000008  x25 0000007997ca75e0  x26 0000007911e14ca0  x27 0000000000000002
    x28 0000000000000000  x29 0000007fd9e97660
    sp  0000007fd9e97620  lr  0000007993182354  pc  000000799318237c

backtrace:
    #00 pc 000000000002237c  /system/lib64/libc.so (abort+116)
    #01 pc 0000000000ceaea0  /data/app/app.omaps.beta-vLqwWAmS4xDBkLPvn4yCSw==/lib/arm64/libmapswithme.so (storage::QueuedCountry::GetRelativeUrl() const+324)
    #02 pc 0000000000ce2128  /data/app/app.omaps.beta-vLqwWAmS4xDBkLPvn4yCSw==/lib/arm64/libmapswithme.so (storage::HttpMapFilesDownloader::Download()+208)
    #03 pc 0000000000ce1fc0  /data/app/app.omaps.beta-vLqwWAmS4xDBkLPvn4yCSw==/lib/arm64/libmapswithme.so (storage::HttpMapFilesDownloader::Download(storage::QueuedCountry&)+256)
    #04 pc 0000000000cea354  /data/app/app.omaps.beta-vLqwWAmS4xDBkLPvn4yCSw==/lib/arm64/libmapswithme.so (storage::Queue::ForEachCountry(std::__ndk1::function<void (storage::QueuedCountry&)> const&)+56)
    #05 pc 0000000000cc3c0c  /data/app/app.omaps.beta-vLqwWAmS4xDBkLPvn4yCSw==/lib/arm64/libmapswithme.so
    #06 pc 00000000007f0974  /data/app/app.omaps.beta-vLqwWAmS4xDBkLPvn4yCSw==/lib/arm64/libmapswithme.so (android::GuiThread::ProcessTask(long)+32)
    #07 pc 0000000000112ff8  /data/app/app.omaps.beta-vLqwWAmS4xDBkLPvn4yCSw==/oat/arm64/base.odex (offset 0x111000) (com.mapswithme.maps.Framework.nativeShowBookmarkCategory [DEDUPED]+152)
    #08 pc 000000000055764c  /system/lib64/libart.so (art_quick_invoke_static_stub+604)
    #09 pc 00000000000cfd14  /system/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+232)
    #10 pc 0000000000280374  /system/lib64/libart.so (art::interpreter::ArtInterpreterToCompiledCodeBridge(art::Thread*, art::ArtMethod*, art::ShadowFrame*, unsigned short, art::JValue*)+344)
    #11 pc 000000000027a388  /system/lib64/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+968)
    #12 pc 0000000000527acc  /system/lib64/libart.so (MterpInvokeStatic+204)
    #13 pc 0000000000549c14  /system/lib64/libart.so (ExecuteMterpImpl+14612)
    #14 pc 00000000009992bc  /data/app/app.omaps.beta-vLqwWAmS4xDBkLPvn4yCSw==/oat/arm64/base.vdex (com.mapswithme.maps.MwmApplication.access$200)
    #15 pc 000000000025408c  /system/lib64/libart.so (_ZN3art11interpreterL7ExecuteEPNS_6ThreadERKNS_20CodeItemDataAccessorERNS_11ShadowFrameENS_6JValueEb.llvm.806176459+488)
    #16 pc 0000000000259b80  /system/lib64/libart.so (art::interpreter::ArtInterpreterToInterpreterBridge(art::Thread*, art::CodeItemDataAccessor const&, art::ShadowFrame*, art::JValue*)+216)
    #17 pc 000000000027a36c  /system/lib64/libart.so (bool art::interpreter::DoCall<false, false>(art::ArtMethod*, art::Thread*, art::ShadowFrame&, art::Instruction const*, unsigned short, art::JValue*)+940)
#18 pc 0000000000527acc  /system/lib64/libart.so (MterpInvokeStatic+204)
    #19 pc 0000000000549c14  /system/lib64/libart.so (ExecuteMterpImpl+14612)
    #20 pc 0000000000999146  /data/app/app.omaps.beta-vLqwWAmS4xDBkLPvn4yCSw==/oat/arm64/base.vdex (com.mapswithme.maps.MwmApplication$1.run+4)
    #21 pc 000000000025408c  /system/lib64/libart.so (_ZN3art11interpreterL7ExecuteEPNS_6ThreadERKNS_20CodeItemDataAccessorERNS_11ShadowFrameENS_6JValueEb.llvm.806176459+488)
    #22 pc 0000000000516e7c  /system/lib64/libart.so (artQuickToInterpreterBridge+1020)
    #23 pc 00000000005604fc  /system/lib64/libart.so (art_quick_to_interpreter_bridge+92)
    #24 pc 000000000176df7c  /system/framework/arm64/boot-framework.oat (offset 0xa2b000) (android.os.Handler.dispatchMessage+76)
    #25 pc 0000000001f66648  /system/framework/arm64/boot-framework.oat (offset 0xa2b000) (android.os.Looper.loop+1304)
    #26 pc 0000000000f11cc8  /system/framework/arm64/boot-framework.oat (offset 0xa2b000) (android.app.ActivityThread.main+664)
    #27 pc 000000000055764c  /system/lib64/libart.so (art_quick_invoke_static_stub+604)
    #28 pc 00000000000cfd14  /system/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+232)
    #29 pc 000000000045de7c  /system/lib64/libart.so (art::(anonymous namespace)::InvokeWithArgArray(art::ScopedObjectAccessAlreadyRunnable const&, art::ArtMethod*, art::(anonymous namespace)::ArgArray*, art::JValue*, char const*)+104)
    #30 pc 000000000045f8d0  /system/lib64/libart.so (art::InvokeMethod(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jobject*, _jobject*, unsigned long)+1440)
    #31 pc 00000000003ef500  /system/lib64/libart.so (art::Method_invoke(_JNIEnv*, _jobject*, _jobject*, _jobjectArray*)+52)
    #32 pc 000000000078eed4  /system/framework/arm64/boot-core-oj.oat (offset 0x2dc000) (java.lang.Class.getDeclaredMethodInternal [DEDUPED]+180)
    #33 pc 000000000244c8d8  /system/framework/arm64/boot-framework.oat (offset 0xa2b000) (com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run+136)
    #34 pc 0000000002458c78  /system/framework/arm64/boot-framework.oat (offset 0xa2b000) (com.android.internal.os.ZygoteInit.main+2104)
    #35 pc 000000000055764c  /system/lib64/libart.so (art_quick_invoke_static_stub+604)
    #36 pc 00000000000cfd14  /system/lib64/libart.so (art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+232)
    #37 pc 000000000045de7c  /system/lib64/libart.so (art::(anonymous namespace)::InvokeWithArgArray(art::ScopedObjectAccessAlreadyRunnable const&, art::ArtMethod*, art::(anonymous namespace)::ArgArray*, art::JValue*, char const*)+104)
    #38 pc 000000000045dadc  /system/lib64/libart.so (art::InvokeWithVarArgs(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jmethodID*, std::__va_list)+424)
    #39 pc 0000000000362d00  /system/lib64/libart.so (art::JNI::CallStaticVoidMethodV(_JNIEnv*, _jclass*, _jmethodID*, std::__va_list)+652)
    #40 pc 00000000000b250c  /system/lib64/libandroid_runtime.so (_JNIEnv::CallStaticVoidMethod(_jclass*, _jmethodID*, ...)+116)
    #41 pc 00000000000b50a8  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::start(char const*, android::Vector<android::String8> const&, bool)+924)
    #42 pc 0000000000002528  /system/bin/app_process64 (main+2012)
    #43 pc 00000000000c8908  /system/lib64/libc.so (__libc_init+88)
```

**Comment by rtsisyk** [comment]
Thanks for your report!

Additional information from Crashlytics:

```
Crashed: Thread :  SIGABRT  0x000027d600005a5b
#00 pc 0x799318237c libc.so 
#01 pc 0x7993182350 libc.so 
#02 pc 0x78f5b30ea0 libmapswithme.so (storage::QueuedCountry::GetRelativeUrl() const [queued_country.cpp:60])
#03 pc 0x78f5b28128 libmapswithme.so (storage::HttpMapFilesDownloader::Download() [http_map_files_downloader.cpp:62])
#04 pc 0x78f5b27fc0 libmapswithme.so (storage::HttpMapFilesDownloader::Download(storage::QueuedCountry&) [http_map_files_downloader.cpp:53])
#05 pc 0x78f5b30354 libmapswithme.so (std::__ndk1::__function::__value_func<void (storage::QueuedCountry&)>::operator()(storage::QueuedCountry&) const [functional:1867])
#06 pc 0x78f5b09c0c libmapswithme.so (storage::MapFilesDownloader::DownloadMapFile(storage::QueuedCountry&)::$_0::operator()()::'lambda'(std::__ndk1::vector<std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char> >, std::__ndk1::allocator<std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char> > > > const&)::operator()(std::__ndk1::vector<std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char> >, std::__ndk1::allocator<std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char> > > > const&) [map_files_downloader.cpp:31])
#07 pc 0x78f5636974 libmapswithme.so (std::__ndk1::__function::__value_func<void ()>::operator()() const [functional:1867])
#08 pc 0x78f7299ff8 base.odex 
```

```
MwmActivity@6d043e7
 8 | Tue Jan 05 2021 11:38:53  | I/MapsmeCore: platform/Platform.cpp:190 Initialize()  Flavor name: web
 9 | Tue Jan 05 2021 11:38:53  | I/MapsmeCore: platform/Platform.cpp:191 Initialize()  Build type name: beta
10 | Tue Jan 05 2021 11:38:53  | I/MapsmeCore: platform/Platform.cpp:215 Initialize()  Apk path =  /data/app/app.omaps.beta-vLqwWAmS4xDBkLPvn4yCSw==/base.apk
11 | Tue Jan 05 2021 11:38:53  | I/MapsmeCore: platform/Platform.cpp:216 Initialize()  Writable path =  /storage/emulated/0/MapsWithMe/
12 | Tue Jan 05 2021 11:38:53  | I/MapsmeCore: platform/Platform.cpp:217 Initialize()  Temporary path =  /storage/emulated/0/Android/data/app.omaps.beta/cache
13 | Tue Jan 05 2021 11:38:53  | I/MapsmeCore: platform/Platform.cpp:218 Initialize()  OBB Google path =  /storage/emulated/0/Android/obb/app.omaps.beta/
14 | Tue Jan 05 2021 11:38:53  | I/MapsmeCore: platform/Platform.cpp:219 Initialize()  OBB Google files =  [0: ]
15 | Tue Jan 05 2021 11:38:53  | I/MapsmeCore:  Loaded countries list for version: 201101
16 | Tue Jan 05 2021 11:38:53  | I/MapsmeCore: tracking/reporter.cpp:88 Run()  Tracking Reporter started
17 | Tue Jan 05 2021 11:38:53  | I/MapsmeCore: editor/user_stats.cpp:78 UserStatsLoader()  There is no cached user stats info in settings
18 | Tue Jan 05 2021 11:38:53  | D/MapsmeCore: indexer/classificator_loader.cpp:47 Load()  Reading of classificator started
19 | Tue Jan 05 2021 11:38:53  | D/MapsmeCore: indexer/classificator_loader.cpp:69 Load()  Reading of classificator finished
20 | Tue Jan 05 2021 11:38:53  | D/MapsmeCore: map/framework.cpp:442 Framework()  Classificator initialized
21 | Tue Jan 05 2021 11:38:53  | D/MapsmeCore: map/framework.cpp:448 Framework()  Country info getter initialized
22 | Tue Jan 05 2021 11:38:53  | D/MapsmeCore: map/framework.cpp:451 Framework()  UGC initialized
23 | Tue Jan 05 2021 11:38:53  | W/MapsmeCore: platform/platform_android.cpp:185 GetReader()  Can't get reader: LocateZipException ../../../../../coding/zip_reader.cpp:56, "Can't locate file inside zip assets/countries-strings/default.json/localize.json"
24 | Tue Jan 05 2021 11:38:53  | W/MapsmeCore: platform/platform_android.cpp:195 GetReader()  Can't get reader for: countries-strings/default.json/localize.json
25 | Tue Jan 05 2021 11:38:53  | W/MapsmeCore: platform/get_text_by_id.cpp:49 GetJsonBuffer()  Can't open default localization file. pathToJson is countries-strings/default.json/localize.json FileAbsentException ../../../../../platform/platform_android.cpp:196, "File not found countries-strings/default.json/localize.json"
26 | Tue Jan 05 2021 11:38:53  | I/MapsmeCore: search/processor.cpp:207 SetPreferredLocale()  New preferred locale: en
27 | Tue Jan 05 2021 11:38:53  | D/MapsmeCore: search/processor.cpp:226 SetInputLocale()  New input locale: en locale code: 1
28 | Tue Jan 05 2021 11:38:53  | I/MapsmeCore: search/processor.cpp:207 SetPreferredLocale()  New preferred locale: ru
29 | Tue Jan 05 2021 11:38:53  | D/MapsmeCore: search/processor.cpp:226 SetInputLocale()  New input locale: ru locale code: 8
30 | Tue Jan 05 2021 11:38:53  | D/MapsmeCore: map/framework.cpp:454 Framework()  Search API initialized
31 | Tue Jan 05 2021 11:38:53  | W/MapsmeCore: search/cities_boundaries_table.cpp:38 Load()  Can't find World map file.
32 | Tue Jan 05 2021 11:38:53  | W/MapsmeCore: search/processor.cpp:358 LoadCitiesBoundaries()  Can't load cities boundaries
33 | Tue Jan 05 2021 11:38:53  | D/MapsmeCore: map/framework.cpp:496 Framework()  Storage initialized
34 | Tue Jan 05 2021 11:38:53  | W/MapsmeCore: platform/platform_android.cpp:185 GetReader()  Can't get reader: LocateZipException ../../../../../coding/zip_reader.cpp:56, "Can't locate file inside zip assets/World.mwm"
35 | Tue Jan 05 2021 11:38:53  | W/MapsmeCore: platform/platform_android.cpp:195 GetReader()  Can't get reader for: World.mwm
```

**Comment by rtsisyk** [comment]
This problem should be fixed on 10.6.0. Please re-open if this issue still exists.
